core/bin/elasticsearch

```diff
@@ -132,120 +121,15 @@ case `uname` in
-        eval exec "$JAVA" $JAVA_OPTS $ES_JAVA_OPTS $es_parms "\"-Des.path.home=$ES_HOME\"" -cp "\"$ES_CLASSPATH\"" $props \
-                org.elasticsearch.bootstrap.Elasticsearch
-        # exec without running it in the background, makes it replace this shell, we'll never get here...
-        # no need to return something
-    else
-        # Startup Elasticsearch, background it, and write the pid.
-        eval exec "$JAVA" $JAVA_OPTS $ES_JAVA_OPTS $es_parms "\"-Des.path.home=$ES_HOME\"" -cp "\"$ES_CLASSPATH\"" $props \
-                    org.elasticsearch.bootstrap.Elasticsearch <&- &
-        return $?
-    fi
-}
-
-# Print command line usage / help
-usage() {
-    echo "Usage: $0 [-vdh] [-p pidfile] [-D prop] [-X prop]"
-    echo "Start elasticsearch."
-    echo "    -d            daemonize (run in background)"
-    echo "    -p pidfile    write PID to <pidfile>"
-    echo "    -h"
-    echo "    --help        print command line options"
-    echo "    -v            print elasticsearch version, then exit"
-    echo "    -D prop       set JAVA system property"
-    echo "    -X prop       set non-standard JAVA system property"
-    echo "   --prop=val"
-    echo "   --prop val     set elasticsearch property (i.e. -Des.<prop>=<val>)"
-}
-
-# Parse any long getopt options and put them into properties before calling getopt below
-# Be dash compatible to make sure running under ubuntu works
-ARGV=""
-while [ $# -gt 0 ]
-do
-    case $1 in
-      --help) ARGV="$ARGV -h"; shift;;
-      --*=*) properties="$properties -Des.${1#--}"
-           shift 1
-           ;;
-      --*) [ $# -le 1 ] && {
-                echo "Option requires an argument: '$1'."
-                shift
-                continue
-            }
-           properties="$properties -Des.${1#--}=$2"
-           shift 2
-           ;;
-      *) ARGV="$ARGV $1" ; shift
-    esac
-done
+export HOSTNAME=`hostname -s`
 
-# Parse any command line options.
-args=`getopt vdhp:D:X: $ARGV`
-eval set -- "$args"
-
-while true; do
-    case $1 in
-        -v)
-            eval "$JAVA" $JAVA_OPTS $ES_JAVA_OPTS $es_parms "\"-Des.path.home=$ES_HOME\"" -cp "\"$ES_CLASSPATH\"" $props \
-                    org.elasticsearch.Version
-            exit 0
-        ;;
-        -p)
-            pidfile="$2"
-            shift 2
-        ;;
-        -d)
-            daemonized="yes"
-            shift
-        ;;
-        -h)
-            usage
-            exit 0
-        ;;
-        -D)
-            properties="$properties -D$2"
-            shift 2
-        ;;
-        -X)
-            properties="$properties -X$2"
-            shift 2
-        ;;
-        --)
-            shift
-            break
-        ;;
-        *)
-            echo "Error parsing argument $1!" >&2
-            usage
-            exit 1
-        ;;
-    esac
-done
-
-# Start up the service
-launch_service "$pidfile" "$daemonized" "$properties"
+daemonized=`echo $* | grep -- -d`
```

Comment:
this only acts on the short option... either find a workaround or also check the long one